pinecone/client_test.go

```diff
@@ -686,11 +581,590 @@ func (ts *ClientTests) TestConfigureIndexHitPodLimit() {
 	require.ErrorContainsf(ts.T(), err, "You've reached the max pods allowed", err.Error())
 }
 
-func (ts *ClientTests) deleteIndex(name string) error {
+func (ts *ClientTestsIntegration) deleteIndex(name string) error {
 	return ts.client.DeleteIndex(context.Background(), name)
 }
 
-// Helper function to check if a name is a valid UUID
+// Unit tests:
+func TestHandleErrorResponseBodyUnit(t *testing.T) {
+	tests := []struct {
+		name         string
+		responseBody *http.Response
+		statusCode   int
+		prefix       string
+		errorOutput  string
+	}{
+		{
+			name:         "test ErrorResponse body",
+			responseBody: mockResponse(`{"error": { "code": "INVALID_ARGUMENT", "message": "test error message"}, "status": 400}`, http.StatusBadRequest),
+			statusCode:   http.StatusBadRequest,
+			errorOutput:  `{"status_code":400,"body":"{\"error\": { \"code\": \"INVALID_ARGUMENT\", \"message\": \"test error message\"}, \"status\": 400}","error_code":"INVALID_ARGUMENT","message":"test error message"}`,
+		}, {
+			name:         "test JSON body",
+			responseBody: mockResponse(`{"message": "test error message", "extraCode": 665}`, http.StatusBadRequest),
+			statusCode:   http.StatusBadRequest,
+			errorOutput:  `{"status_code":400,"body":"{\"message\": \"test error message\", \"extraCode\": 665}"}`,
+		}, {
+			name:         "test string body",
+			responseBody: mockResponse(`test error message`, http.StatusBadRequest),
+			statusCode:   http.StatusBadRequest,
+			errorOutput:  `{"status_code":400,"body":"test error message"}`,
+		}, {
+			name:         "Test error response with empty response",
+			responseBody: mockResponse(`{}`, http.StatusBadRequest),
+			statusCode:   http.StatusBadRequest,
+			prefix:       "test prefix",
+			errorOutput:  `{"status_code":400,"body":"{}"}`,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			err := handleErrorResponseBody(tt.responseBody, tt.prefix)
+			assert.Equal(t, err.Error(), tt.errorOutput, "Expected error to be '%s', but got '%s'", tt.errorOutput, err.Error())
+
+		})
+	}
+}
+
+func TestFormatErrorUnit(t *testing.T) {
+	tests := []struct {
+		name     string
+		err      int
+		expected *PineconeError
+	}{
+		{ // TODO: should really add a negative case here, but triggering a marshalling error is hard
```

Comment:
Just curious, what would a negative case be for this?

Reply:
uhh idk , ignore this, apologies

Reply:
It's no problem, was just curious